android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -268,6 +268,13 @@ public void onUrlDeviceDiscovered(UrlDevice urlDevice) {
     // Don't short circuit because icons
     // and metadata may not be fetched
     mPwCollection.addUrlDevice(urlDevice);
+    if (!Utils.isPublic(urlDevice)) {
```

Comment:
"isPublic" is currently used to determine whether or not the result should be private to the user (for example, would it be OK to show on the lock screen if they don't want private data displayed on the lock screen).  This is different from the concept of a URL being available only on the local network.  Instead, just make an isLocalDevice() function that checks for title & description.